builder/dockerfile/builder.go

```diff
@@ -240,24 +240,27 @@ func (b *Builder) build(source builder.Source, dockerfile *parser.Result) (*buil
 
 	addNodesForLabelOption(dockerfile.AST, b.options.Labels)
 
-	if err := checkDispatchDockerfile(dockerfile.AST); err != nil {
-		buildsFailed.WithValues(metricsDockerfileSyntaxError).Inc()
-		return nil, err
-	}
-
-	dispatchState, err := b.dispatchDockerfileWithCancellation(dockerfile, source)
+	stages, metaArgs, err := instructions.Parse(dockerfile.AST)
 	if err != nil {
+		if instructions.IsUnknownInstruction(err) {
+			buildsFailed.WithValues(metricsUnknownInstructionError).Inc()
+		}
```

Comment:
There was some discussion of changing this to be a more generic "parse failed" error so that we don't need this condition. What was the outcome of that?